test/e2e-techpreview/onclusterlayering_test.go

```diff
@@ -401,7 +401,7 @@ func TestDeletedTransientMachineOSBuildIsRecreated(t *testing.T) {
 	// Wait for the build to start
 	firstMosb := waitForBuildToStartForPoolAndConfig(t, cs, poolName, mosc.Name)
 
-	firstPod, err := cs.CoreV1Interface.Pods(ctrlcommon.MCONamespace).Get(ctx, utils.GetBuildPodName(firstMosb), metav1.GetOptions{})
+	firstJob, err := cs.BatchV1Interface.Jobs(ctrlcommon.MCONamespace).Get(ctx, utils.GetBuildJobName(firstMosb), metav1.GetOptions{})
```

Comment:
thought (non-blocking): Another interesting test case might be to delete the actual build pod (as opposed to the higher-level Job). Ideally, the MachineOSBuild should not be considered interrupted in that case since the Job controller should reschedule a new pod.

Reply:
Added a test case for this